code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+		qdel(B)
+
+		blood_amt++
+		if(blood_amt == max_blood_amt)
+			return
+
+/obj/structure/slime_crystal/red/attack_hand(mob/user)
+	if(blood_amt < 100)
+		return ..()
+
+	blood_amt -= 100
+	var/type = pick(/obj/item/reagent_containers/food/snacks/meat/slab,/obj/item/organ/heart,/obj/item/organ/lungs,/obj/item/organ/liver,/obj/item/organ/eyes,/obj/item/organ/tongue,/obj/item/organ/stomach,/obj/item/organ/ears)
+	new type(get_turf(src))
+
+/obj/structure/slime_crystal/red/attacked_by(obj/item/I, mob/living/user)
+	if(blood_amt < 10)
+		return ..()
+
+	if(!istype(I,/obj/item/reagent_containers/glass/beaker))
+		return ..()
+
+	var/obj/item/reagent_containers/glass/beaker/item_beaker = I
+
+	if(!item_beaker.is_refillable() || (item_beaker.reagents.total_volume + 10 > item_beaker.reagents.maximum_volume))
+		return ..()
+	blood_amt -= 10
+	item_beaker.reagents.add_reagent(/datum/reagent/blood,10)
+
+/obj/structure/slime_crystal/green
+	colour = "green"
+	var/datum/mutation/stored_mutation
+
+/obj/structure/slime_crystal/green/examine(mob/user)
+	. = ..()
+	if(stored_mutation)
+		. += "It currently stores [stored_mutation.name]"
+	else
+		. += "It doesn't hold any mutations"
+
+/obj/structure/slime_crystal/green/attack_hand(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_user = user
+	var/list/mutation_list = human_user.dna.mutations
+	stored_mutation = pick(mutation_list)
+	stored_mutation = stored_mutation.type
+
+/obj/structure/slime_crystal/green/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!ishuman(affected_mob) || !stored_mutation)
+		return
+	var/mob/living/carbon/human/human_mob = affected_mob
+	human_mob.dna.add_mutation(stored_mutation)
+	if(affected_mobs[affected_mob] % 60 == 0)
+		return
+
+	var/list/mut_list = human_mob.dna.mutations
+	var/list/secondary_list = list()
+
+	for(var/X in mut_list)
+		if(istype(X,stored_mutation))
+			continue
+		var/datum/mutation/t_mutation = X
+		secondary_list += t_mutation.type
+
+	var/datum/mutation/mutation = pick(secondary_list)
+	human_mob.dna.remove_mutation(mutation)
+
+/obj/structure/slime_crystal/green/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	if(!ishuman(affected_mob))
+		return
+	var/mob/living/carbon/human/human_mob = affected_mob
+	human_mob.dna.remove_mutation(stored_mutation)
+
+/obj/structure/slime_crystal/pink
+	colour = "pink"
+
+/obj/structure/slime_crystal/pink/on_mob_enter(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob,TRAIT_PACIFISM,MAGIC_TRAIT)
+
+/obj/structure/slime_crystal/pink/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	REMOVE_TRAIT(affected_mob,TRAIT_PACIFISM,MAGIC_TRAIT)
+
+/obj/structure/slime_crystal/gold
+	colour = "gold"
+
+/obj/structure/slime_crystal/gold/attack_hand(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_mob = user
+	var/mob/living/simple_animal/pet/chosen_pet = pick(subtypesof(/mob/living/simple_animal/pet))
```

Comment:
Should ignore unique pet types such as Ian which do extra things and interact with persistance